clang/test/Preprocessor/cheri-lock-free.c

```diff
@@ -0,0 +1,37 @@
+/// Check that we report pointers as being always lock-free, otherwise <atomic>
+/// ends up using locks with -ffreestanding.
+// RUN: %riscv64_cheri_cc1 -fgnuc-version=4.2.1 -target-feature +a -E -dM %s \
+// RUN:    | FileCheck %s --check-prefixes=CHECK,CHECK-64 --implicit-check-not=_LOCK_FREE
+// RUN: %riscv64_cheri_purecap_cc1 -fgnuc-version=4.2.1 -target-feature +a -E -dM %s \
+// RUN:    | FileCheck %s --check-prefixes=CHECK,CHECK-64 --implicit-check-not=_LOCK_FREE
+// RUN: %riscv32_cheri_cc1 -fgnuc-version=4.2.1 -target-feature +a -E -dM %s \
+// RUN:    | FileCheck %s --check-prefixes=CHECK,CHECK-32 --implicit-check-not=_LOCK_FREE
+// RUN: %riscv32_cheri_purecap_cc1 -fgnuc-version=4.2.1 -target-feature +a -E -dM %s \
+// RUN:    | FileCheck %s --check-prefixes=CHECK,CHECK-32 --implicit-check-not=_LOCK_FREE
+// RUN: %cheri128_cc1 -fgnuc-version=4.2.1 -E -dM %s \
+// RUN:    | FileCheck %s --check-prefixes=CHECK,CHECK-64 --implicit-check-not=_LOCK_FREE
+// RUN: %cheri128_purecap_cc1 -fgnuc-version=4.2.1 -E -dM %s \
+// RUN:    | FileCheck %s --check-prefixes=CHECK,CHECK-64 --implicit-check-not=_LOCK_FREE
+
+// CHECK: #define __CLANG_ATOMIC_BOOL_LOCK_FREE 2
+// CHECK: #define __CLANG_ATOMIC_CHAR16_T_LOCK_FREE 2
+// CHECK: #define __CLANG_ATOMIC_CHAR32_T_LOCK_FREE 2
+// CHECK: #define __CLANG_ATOMIC_CHAR_LOCK_FREE 2
+// CHECK: #define __CLANG_ATOMIC_INT_LOCK_FREE 2
+// CHECK-64: #define __CLANG_ATOMIC_LLONG_LOCK_FREE 2
+// CHECK-32: #define __CLANG_ATOMIC_LLONG_LOCK_FREE 1
```

Comment:
I guess we could use capability atomics to make those operations lock-free, but I'm not sure if we should.